examples/sklearnex/spmd_knn_bf_regression.py

```diff
@@ -0,0 +1,69 @@
+#===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+import numpy as np
+from warnings import warn
+from mpi4py import MPI
+import dpctl
+from numpy.testing import assert_allclose
+from sklearnex.spmd.neighbors import KNeighborsRegressor as KnnRegSpmd
+
+
+def generate_X_y(par, coef_seed, data_seed):
+    ns, nf = par['ns'], par['nf']
+
+    crng = np.random.default_rng(coef_seed)
+    coef = crng.uniform(-10, 10, size=(nf,))
+
+    drng = np.random.default_rng(data_seed)
+    data = drng.uniform(-100, 100, size=(ns, nf))
+    resp = data @ coef
+
+    return data, resp, coef
+
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+
+if size < 2:
+    warn("This example was intentionally "
+         "designed to run in distributed mode only", RuntimeWarning)
+
+q = dpctl.SyclQueue("gpu")
```

Comment:
GPU unavailability should be handled. For example, use exit with warning

Reply:
How does this seem - https://github.com/ethanglaser/scikit-learn-intelex/blob/dev/eglaser-spmd-knn/examples/sklearnex/spmd_knn_bf_regression.py#L46-L49

Reply:
I meant it should have better explanation why example can't be run. dpctl itself produces error saying 'device is unavailable'. @KulikovNikita, what do think about adding warning `GPU device is unavailable. Currently, SPMD execution mode is implemented only for this device type.`